learning/tour-of-beam/frontend/lib/main.dart

```diff
@@ -46,11 +49,20 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      localizationsDelegates: context.localizationDelegates,
-      supportedLocales: context.supportedLocales,
-      locale: context.locale,
-      home: const HomeScreen(),
+    return ThemeSwitchNotifierProvider(
+      child: Consumer<ThemeSwitchNotifier>(
+        builder: (context, themeSwitchNotifier, _) {
+          return MaterialApp(
```

Comment:
I like puttnig `MaterialApp` as high as possible. Otherwise, when we switch to a router, it may be doing some initialization that the upper widgets will race with.

Reply:
Can it get higher than ThemeSwitchNotifierProvider?

Reply:
Oh, I guess not. Thanks.